roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -242,6 +244,81 @@ object rblstring {
     }
   }
 
+  /**
+    * Define the string-split primitive.
+    * This returns a Tuple containing tokens from a string.
+    * Split is done at the specified separators.
+    * An optional maximum number of splits may be specified
+    * Leading and trailing separators are ignored.
+    * if no speparators, return a Tuple of the string's characters ignoring the count
+    *
+    * examples:
+    * (string-split "a,b,c,d,e,f" ",") ====> ["a" "b" "c" "d" "e" "f" #niv]
+    * (string-split "a,b.c,d.e,f" "," 4) ====> ["a" "b.c" "d.e" "f" #niv]
+    * (string-split "a,b.c,d.e,f" "," 0) ====> []
+    * (string-split "a,b.c,d.e,f" "," -1) ==== []
+    * (string-split ",a,b,c," "," 2) ====> ["a" "b" "c,"]
+    * (string-split ",a,b,c," "," 1) ====> ["a" "b,c,"]
+    * (string-split ",a,b,c," ",") ====> ["a" "b" "c" #niv]
+    * (string-split "a,b,c" "") ====> [#\a #\, #\b #\, #\c]
+    * (string-split "a,b,c,d,e,f" "," 3) ====> ["a" "b" "c" "d,e,f"]
+    * (string-split "a,b.c,d.e,f" "," 2) ====> ["a" "b.c" "d.e,f"]
+    * (string-split "a,b.c,d.e,f" ",." 2) ====> ["a" "b" "c,d.e,f"]
+    * (string-split "a,b.c,d.e,f" "," 1) ====> ["a" "b.c,d.e,f"]
+    */
+  object stringSplit extends Prim {
+    override val name: String = "string-split"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 3
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
+      val elem = ctxt.argvec.elem
+
+      for {
+        str <- checkType[RblString](0, elem) // Ensure arg0 is a RblString
+        sep <- checkType[RblString](1, elem) // Ensure arg1 is a RblString
+        w <- checkType[Fixnum](2, elem) // Ensure optional arg2 is a Fixnum
+            match {
+              case Right(v)                 => Right(v)
+              case Left(TypeMismatch(_, _)) => Left(TypeMismatch(2, Fixnum.getClass.getName))
+              case Left(_)                  => Right(Fixnum(Int.MaxValue))
+            }
+      } yield {
+
+        // At this point:
+        //  str = string to tokenize
+        //  sep = string of separators
+        //  w = count of tokens to return
+
+        val tl = // Get the list of tokens
+          if (sep.value.isEmpty) // Don't upset split with an empty list of separators
+            Array.empty[String]
+          else
+            str.value.split(sep.value.toArray) // split ignores trailing separators
+
+        val tr = tl.dropWhile(_.isEmpty) // Trim off leading separators
+
+        if (sep.value.isEmpty) { // no separators? return a Tuple of RblChars
+          Tuple(str.value.toCharArray.toSeq.map(RblChar(_)))
```

Comment:
Remove `.toCharArray` it's not necessary

Reply:
Done